packages/facility-server/app/routes/apiv1/patient/patientprogramRegistration/patientProgramRegistration.js

```diff
@@ -111,6 +93,59 @@ patientProgramRegistration.post(
   }),
 );
 
+patientProgramRegistration.put(
+  '/programRegistration/:programRegistrationId',
+  asyncHandler(async (req, res) => {
+    req.checkPermission('write', 'PatientProgramRegistration');
+    const { db, models, params, body } = req;
+    const { programRegistrationId } = params;
+    const { conditions = [], ...registrationData } = body;
+
+    if (conditions.length > 0) {
+      req.checkPermission('create', 'PatientProgramRegistrationCondition');
+    }
+
+    const existingRegistration =
+      await models.PatientProgramRegistration.findByPk(programRegistrationId);
+
+    if (!existingRegistration) {
+      throw new NotFoundError('PatientProgramRegistration not found');
+    }
+
+    const { patientId, programRegistryId } = existingRegistration;
+
+    const conditionsData = conditions
+      .filter((condition) => condition.id)
+      .map((condition) => ({
+        id: condition.id,
+        patientId,
+        programRegistryId,
+        clinicianId: registrationData.clinicianId,
+        date: condition.date,
+        programRegistryConditionId: condition.conditionId,
+        conditionCategory: condition.conditionCategory,
+        reasonForChange: condition.reasonForChange,
+      }));
+
+    const [registration, conditionsRecords] = await db.transaction(async () => {
+      return Promise.all([
+        existingRegistration.update(body),
+        models.PatientProgramRegistrationCondition.bulkCreate(conditionsData, {
+          updateOnDuplicate: ['date', 'conditionCategory', 'reasonForChange'],
+        }),
+      ]);
+    });
+
+    // Convert Sequelize model to use a custom object as response
+    const responseObject = {
+      ...registration.get({ plain: true }),
+      conditions: conditionsRecords,
+    };
```

Comment:
Not entirely sure what the expected behaviour would be here. With the behaviour as it is currently, if the conditions weren't changed we'd return with a `conditions: []` field which feels a little misleading.

Thoughts on just returning the `registration`? Or returning registration and conditions if conditions have been updated?

Reply:
You are right - I think it should return the registration record with the conditions attached or just the registration record by itself. I decided on the simple record option as we aren't doing anything with this response so it seems unnecessary to have an extra DB query and have extra code to maintain.